sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -355,4 +359,63 @@ public void removeExtra(@NotNull String key) {
   public void addEventProcessor(@NotNull EventProcessor eventProcessor) {
     eventProcessors.add(eventProcessor);
   }
+
+  // Atomic operations on session
+  public void withSession(@NotNull IWithSession sessionCallback) {
```

Comment:
Ideally something like:

```suggestion
  @Api.Internal
  public void withSession(@NotNull IWithSession sessionCallback) {
```